File: domain/src/main/java/com/balv/imdb/domain/usecases/GetMovieListUseCase.java
package com.balv.imdb.domain.usecases;

import com.balv.imdb.domain.models.Movie;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Observable;
import kotlin.Unit;

public class GetMovieListUseCase extends BaseUseCase<Unit, List<Movie>> {
    @Inject
    public GetMovieListUseCase(){}
    @Override
    public Observable<List<Movie>> execute(Unit input) {
        return mMovieRepo.getLocalMovieData();
    }
}

File: domain/src/main/java/com/balv/imdb/domain/usecases/GetMovieListUseCase.kt
package com.balv.imdb.domain.usecases

import androidx.paging.PagingData
import com.balv.imdb.domain.models.Movie
import com.balv.imdb.domain.repositories.IMovieRepository
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

class GetMovieListUseCase @Inject constructor(
    mMovieRepo: IMovieRepository
) : BaseUseCase<Unit, Flow<PagingData<Movie>>>(mMovieRepo) {
    override suspend fun execute(input: Unit): Flow<PagingData<Movie>> {
        return mMovieRepo.allMoviesPaging()
    }
}
